test/integration/helmcharts/chartstests.go

```diff
@@ -0,0 +1,211 @@
+			utils.ExpectNoError(utils.WaitForDeploymentToBeReady(ctx, f.TestLog(), f.Client, kutil.ObjectKey(deployerName, state.Namespace), 2*time.Minute))
+		})
+
+		It("should deploy the Manifest deployer through its Helm Chart", func() {
+			ctx := context.Background()
+			defer ctx.Done()
+
+			state, cleanup, err := f.NewState(ctx)
+			utils.ExpectNoError(err)
+			dumper.AddNamespaces(state.Namespace)
+			defer func() {
+				Expect(cleanup(ctx)).ToNot(HaveOccurred())
+			}()
+
+			By("Setting up the Helm deployer first")
+			deployHelmDeployer(ctx, state, f, dumper)
+
+			By("Creating and applying a Manifest Deployer DeployItem")
+			var (
+				chartDir   = filepath.Join(f.RootPath, "/charts/manifest-deployer")
+				valuesFile = filepath.Join(chartDir, "values.yaml")
+			)
+
+			const deployerName = "manifest-deployer"
+
+			target, err := utils.CreateInternalKubernetesTarget(ctx, f.Client, state.Namespace, deployerName, f.RestConfig)
+			utils.ExpectNoError(err)
+			utils.ExpectNoError(state.Create(ctx, f.Client, target))
+
+			di := forgeDeployItem(chartDir, valuesFile, deployerName, target)
+			utils.ExpectNoError(state.Create(ctx, f.Client, di))
+			utils.ExpectNoError(utils.WaitForHelmDeployItemToSucceed(ctx, f.TestLog(), f.Client, kutil.ObjectKey(di.Name, state.Namespace), 2*time.Minute))
+			utils.ExpectNoError(utils.WaitForDeploymentToBeReady(ctx, f.TestLog(), f.Client, kutil.ObjectKey(deployerName, state.Namespace), 2*time.Minute))
+		})
+
+		It("should deploy the Mock deployer through its Helm Chart", func() {
+			ctx := context.Background()
+			defer ctx.Done()
+
+			state, cleanup, err := f.NewState(ctx)
+			utils.ExpectNoError(err)
+			dumper.AddNamespaces(state.Namespace)
+			defer func() {
+				Expect(cleanup(ctx)).ToNot(HaveOccurred())
+			}()
+
+			By("Setting up the Helm deployer first")
+			deployHelmDeployer(ctx, state, f, dumper)
+
+			By("Creating and applying a Mock Deployer DeployItem")
+			var (
+				chartDir   = filepath.Join(f.RootPath, "/charts/mock-deployer")
+				valuesFile = filepath.Join(chartDir, "values.yaml")
+			)
+
+			const deployerName = "mock-deployer"
+
+			target, err := utils.CreateInternalKubernetesTarget(ctx, f.Client, state.Namespace, deployerName, f.RestConfig)
+			utils.ExpectNoError(err)
+			utils.ExpectNoError(state.Create(ctx, f.Client, target))
+
+			di := forgeDeployItem(chartDir, valuesFile, deployerName, target)
+			utils.ExpectNoError(state.Create(ctx, f.Client, di))
+			utils.ExpectNoError(utils.WaitForHelmDeployItemToSucceed(ctx, f.TestLog(), f.Client, kutil.ObjectKey(di.Name, state.Namespace), 2*time.Minute))
+			utils.ExpectNoError(utils.WaitForDeploymentToBeReady(ctx, f.TestLog(), f.Client, kutil.ObjectKey(deployerName, state.Namespace), 2*time.Minute))
+		})
+	})
+}
+
+func deployHelmDeployer(ctx context.Context, state *envtest.State, f *framework.Framework, dumper *framework.Dumper) {
```

Comment:
the helm deployer is installed by creating a deployitem of type helm with the helmdeployer as base64 encoded chart and then call the helm specific go code to deploy the item?

Reply:
Chicken-Egg problem. We need the Helm deployer to deploy Helm charts so how do we get the Helm deployer into the cluster?
However, this was based on the assumption that we do not have a landscaper-controller with integrated deployers in the test cluster (as it was the case for my local setup). If we we have the landscaper-controller with enabled internal helm deployer available, then this code is indeed not needed.

Reply:
currently it is like that. And I think we also planned afterwards to keep a lightwight version. 
So lets keep it simple and expect a running deployer

Reply:
Fixed, the code that deploys the initial Helm-Deployer is removed and the landscaper-controller's internal Helm-deployer is now responsible for picking up the DeployItems.
The Deployers that get deployed during the test will have an active TargetSelector so that they will not pick up any DeployItem (unless they have a matching annotation of course).